substrate/frame/revive/rpc/src/receipt_provider.rs

```diff
@@ -256,19 +256,26 @@ impl<B: BlockInfoProvider> ReceiptProvider<B> {
 		block_number: SubstrateBlockNumber,
 		block_map: &BlockHashMap,
 	) -> Result<(), ClientError> {
-		// Keep track of the latest block hashes, so we can prune older blocks.
-		if let Some(keep_latest_n_blocks) = self.keep_latest_n_blocks {
-			let mut block_number_to_hash = self.block_number_to_hashes.lock().await;
-
-			// Fork? - If inserting the same block number with a different hash, remove the old one
-			let mut to_remove = Vec::new();
-			match block_number_to_hash.insert(block_number, block_map.clone()) {
-				Some(old_block_map) if &old_block_map != block_map => {
+		let mut to_remove = Vec::new();
+		let mut block_number_to_hash = self.block_number_to_hashes.lock().await;
+
+		// Fork? - If inserting the same block number with a different hash, remove the old ones.
+		match block_number_to_hash.insert(block_number, block_map.clone()) {
+			Some(old_block_map) if &old_block_map != block_map => {
+				to_remove.push(old_block_map);
+
+				// Now loop through the blocks that were building on top of the old fork and remove
```

Comment:
Out of curiosity. With the old approach wouldn't the block from the old fork be removed sequentially upon  insertions of the block from the new fork?
eg. 
1. block 4 to be inserted --> remove block 4 from old fork
2. block 5 to be inserted --> remove block 5 from old fork

Reply:
if for example you do a chain reversion at block 4 (dropping blocks 4 and 5).
Then you rebuild block 4, but it includes a transaction hash from block 5. It will fail to add it because the transaction_hash is a primary key and therefore wouldn't be unique